New detector naive WIP

breznak opened the pull request:
WIP 

This is based on #341 

- [x] implement naive predictor
- [ ] how to obtain anomaly score from scalars predicted & current? 

Fixes #343

breznak commented on nab/detectors/naive/naive_detector.py:
```diff
@@ -0,0 +1,46 @@
+    """The anomaly score is simply the last seen value"""
+    current = float(inputData["value"])
+    inputData['predicted'] = self.predicted
+    anomalyScore = 1 #FIXME how compute anomaly score from predicted, current?
```
Looking for help: this is the implementation of naive predictor. 
How do I compute anomaly score from the 2 numbers?

Japillow replied:
You could use something like 1-exp(-nx) where n is a hyperparameter (for ex n = 10) and x = abs(current - predicted)/predicted. Other options could be arctan, 1-n/(x+1), etc

Japillow commented on nab/detectors/naive/naive_detector.py:
```diff
@@ -25,12 +25,22 @@
+    self.predicted = 0.0 #previous value, last seen
 
   def handleRecord(self, inputData):
     """The anomaly score is simply the last seen value"""
```
change `anomaly score` to `predicted value`

breznak commented:
> You could use something like 1-exp(-nx) where n is a hyperparameter (for ex n = 10) and x = abs(current - predicted)/predicted. 

Thank you very much, @Japillow ! I've implemented the function, needed to do small adaptations for numeric stability. Could you have a look, please? 

Updated the doc. 

PS: the results are very bad: 0/0/0, I'm not sure if it's in the implementation, the params of the anomaly function, or just that the "naive" predictor does not work that good on these datasets?

Japillow commented on nab/detectors/naive/naive_detector.py:
```diff
@@ -18,29 +18,74 @@
+    x = abs(current - predicted)/predicted
+    score = 1-math.exp(-self.coef * x)
+
+    # bound to anomaly range (should not happen, but some are over)
```
If some are over (or below) there must be an error somewhere, though I can't find it. For which values does this happen ?

breznak replied:
> must be an error somewhere, though I can't find it. For which values does this happen ? 

I'll add debug prints. Couldn't get it properly, because the benchmark runs multithreaded.

breznak commented:
@Japillow could you please help me finish this PR?
